### PR TITLE
XRT-844 & XRT-850

### PR DESCRIPTION
**Pull request by AShivangi**
### Work done:
- XRT-844 Regression: port xbutil validate new flow to xbutil2
- XRT-850 u30 specific SC flash port to xbmgmt2

### Test Matrix

|   | Legacy flow (python) (u200) | New flow (c++ with platform.json)  (u30)|
| ------------- | ------------- |------------- |
| **Ubuntu 16.04 (no boost process)**  | :white_check_mark: | :white_check_mark:  |
| **Ubuntu 18.04 (boost process)**  |  :white_check_mark:  |  :white_check_mark: |

### CODE REVIEW

File: src/runtime_src/core/tools/xbutil2/SubCmdValidate.cpp [hunk 1]
```diff
@@ -264,30 +264,83 @@ runTestCase(const std::shared_ptr<xrt_core::device>& _dev, const std::string& py
   }
   // log xclbin path for debugging purposes
   logger(_ptTest, "Xclbin", xclbinPath);
+  auto json_exists = [xclbinPath]() {
+    boost::filesystem::path test_dir(xclbinPath);
+    return boost::filesystem::exists(test_dir.parent_path().string() + "/platform.json") ? true : false;
```
Comment (rozumx) [inline src/runtime_src/core/tools/xbutil2/SubCmdValidate.cpp:269]: Magic string.  Please define and use the string and use boost::filesystem::path to join the paths.  For example:
```
static std::string PLATFORM_METADATA = "platform.json";
boost::filesystem::path platform_file(xclbinPath);
platform_file=/ PLATFORM_METADATA;
return boost::filesystem::exists(platform_file) ? true : false;
```

File: src/runtime_src/core/tools/xbutil2/SubCmdValidate.cpp [hunk 2]
```diff
@@ -264,30 +264,83 @@ runTestCase(const std::shared_ptr<xrt_core::device>& _dev, const std::string& py
...
+      logger(_ptTest, "Error", "Please check if the platform package is installed correctly");
+      _ptTest.put("status", "failed");
+      return;
+    }
+
+    // log testcase path for debugging purposes
+    logger(_ptTest, "Testcase", xrtTestCasePath);
+
+    boost::filesystem::path test_dir(xclbinPath);
+    std::vector<std::string> args = { test_dir.parent_path().string(), "-d",
```
Comment (rozumx) [inline src/runtime_src/core/tools/xbutil2/SubCmdValidate.cpp:302]: Pretty the format
```
std::vector<std::string> args = { test_dir.parent_path().string(), 
                                  "-d", xrt_core::query::pcie_bdf::to_string(xrt_core::device_query<xrt_core::query::pcie_bdf>(_dev)) };
```

File: src/runtime_src/core/tools/xbutil2/SubCmdValidate.cpp [hunk 3]
```diff
@@ -264,30 +264,83 @@ runTestCase(const std::shared_ptr<xrt_core::device>& _dev, const std::string& py
...
+    if (!boost::filesystem::exists(xrt_path)) {
+      logger(_ptTest, "Error", boost::str(boost::format("Failed to find %s") % xrtTestCasePath));
+      logger(_ptTest, "Error", "Please check if the platform package is installed correctly");
+      _ptTest.put("status", "failed");
+      return;
+    }
+    // log testcase path for debugging purposes
+    logger(_ptTest, "Testcase", xrtTestCasePath);
+
+    std::vector<std::string> args = { "-k", xclbinPath, "-d", std::to_string(_dev.get()->get_device_id()) };
```
Comment (rozumx) [inline src/runtime_src/core/tools/xbutil2/SubCmdValidate.cpp:330]: Pretty the format
```
std::vector<std::string> args = { "-k", xclbinPath, 
                                  "-d", std::to_string(_dev.get()->get_device_id()) };
```

**Review by rozumx — APPROVED** [review]
Minor format changes and code refactoring.

File: src/runtime_src/core/tools/xbutil2/SubCmdValidate.cpp (hunk 1 above)
Comment (AShivangi) [reply src/runtime_src/core/tools/xbutil2/SubCmdValidate.cpp]: Fixed

File: src/runtime_src/core/tools/xbutil2/SubCmdValidate.cpp (hunk 3 above)
Comment (AShivangi) [reply src/runtime_src/core/tools/xbutil2/SubCmdValidate.cpp]: fixed

File: src/runtime_src/core/tools/xbutil2/SubCmdValidate.cpp (hunk 2 above)
Comment (AShivangi) [reply src/runtime_src/core/tools/xbutil2/SubCmdValidate.cpp]: fixed

**Comment by xilinxgitops** [comment]
retest this please

**Pull request merged**